src/util.c

```diff
@@ -120,7 +120,6 @@ static int stringmatchlen_impl(const char *pattern,
                 if (patternLen >= 2 && pattern[0] == '\\') {
                     pattern++;
                     patternLen--;
-                    if (pattern[0] == string[0]) match = 1;
```

Comment:
Are there no escapes special characters in a character ranges? Here it looks like you just skip all backslashes. What about patterns like `[\]]`, `[\\]`, `[\^]` and `[a\-z]`?

To me it looks like this line is needed, but it should be case-insensitive if nocase is given.

Reply:
Ah, right.